sycl/doc/extensions/proposed/sycl_ext_oneapi_kernel_compiler_spirv.asciidoc

```diff
@@ -0,0 +1,278 @@
+ feature-test macro always has this value.
+|===
+
+=== New source language enumerator
+
+This extension adds the `spirv` enumerator to the `source_language`
+enumeration, which indicates that a kernel bundle defines kernels as a SPIR-V
+binary module.
+
+```
+namespace sycl::ext::oneapi::experimental {
+
+enum class source_language : /*unspecified*/ {
+  // ...
+  spirv
+};
+
+} // namespace sycl::ext::oneapi::experimental
+```
+
+=== Build options
+
+This extension does not specify any options that may be passed via the
+`build_options` property, however an implementation may allow
+implementation-defined options to be passed this way.
+
+=== SPIR-V execution environment
+
+The precise rules for interpreting a SPIR-V module are defined both by the
+SPIR-V specification and by the SPIR-V execution environment.
+For the purposes of this SYCL extension, the SPIR-V execution environment is
+defined by the SYCL backend.
+Each SYCL backend that supports this extension must provide a SPIR-V client API
+specification that formally defines the SPIR-V execution environment.
+
+[_Note:_ {dpcpp} provides two backends that can support this SYCL extension,
+and each has its own SPIR-V client API.
+For the Level Zero backend, see the
+https://spec.oneapi.io/level-zero/latest/core/SPIRV.html[SPIR-V Programming
+Guide].
+For the OpenCL backend, see the
+https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_Env.html[
+OpenCL SPIR-V Environment Specification].
+_{endnote}_]
+
+=== Obtaining a kernel
+
+In SPIR-V, a kernel is represented as an *OpEntryPoint*, which has a literal
+_Name_ parameter.
+The `ext_oneapi_has_kernel` and `ext_oneapi_get_kernel` member functions
+identify a kernel using the name, exactly as it appears in the *OpEntryPoint*
+instruction.
+For example, if the kernel is defined this way in SPIR-V:
+
+```
+OpEntryPoint Kernel %20 "foo"
+```
+
+Then the application's host code can query for the kernel like so:
+
+```
+sycl::kernel_bundle<sycl::bundle_state::executable> kb = /*...*/;
+sycl::kernel k = kb.ext_oneapi_get_kernel("foo");
+```
+
+There is no mechanism to decorate a SPIR-V kernel with a type-name, so the
+forms of `kernel_bundle::has_kernel` or `kernel_bundle::get_kernel` that take a
+type-name are not useful for kernels defined in SPIR-V.
+
+=== Passing kernel arguments
+
+The SPIR-V *OpEntryPoint* that defines a kernel has an associated *OpFunction*.
+This *OpFunction* is followed by a list of *OpFunctionParameter* instructions,
+one for each kernel argument.
+The following table defines the set of argument types that are supported by
+this extension, and explains how to pass each type of argument from SYCL.
+However, the set of supported argument types may be further limited by the
+backend's SPIR-V client API specification.
+
+[%header,cols="1,1"]
+|===
+|SPIR-V type
+|Corresponding SYCL type
+
+|*OpTypeInt*
+|A C++ integral type with the same width.
+
+|*OpTypeFloat* of width 32
+|`float`
```

Comment:
Is there any reason why dtypes are mentioned here? As long as it is supported by SPIR-V, it should be fine.
There are other dtypes that are important for us and they are not mentioned i.e. bfloat16 or float16/half